app/assets/javascripts/sitewide/print_progress_modal.js

```diff
@@ -0,0 +1,24 @@
+$(document).on('click', '.label-printing-progress-modal .close', function() {
+  $(this).parent().parent().remove();
+});
+
+$(document).on('turbolinks:load', function() {
+  var modal = $(document).find('.label-printing-progress-modal');
+  if (modal.length > 0) {
+    window.printerModalReloadInterval = setInterval(function() {
```

Comment:
Use a recursive function with setTimeout. It will be better from point of performance.